### DISCUSSION TITLE
PostgreSQL and Materialized Views - do You need my help maybe? :)

### DISCUSSION THREAD

**Discussion by tztztztz** [phalcon/cphalcon#15464] [Ideas]
Hello guys,

I had a problem with Phalcon 4, PostgreSQL and Materialized Views. 

Phalcon 4 doesn't recognise Materialized Views as they are not present in "information_schema.columns" View.

I've resolved it by:

1) creating dedicated view that mimics "information_schema.columns", I did it by manipulating SQL View that creates it, see original SQL of the View: select pg_get_viewdef('information_schema.columns', true);

2) I've created my own SQL Dialect and SQL Adapter for Postgresql that uses my view instead of "information_schema.columns" view.

Now I can create \Phalcon\Mvc\Model class that will work with Materialized Views.

I wonder if anyone needs that, if so there are two ways to implement that functionality:

1) Create guide in documentation, where whole thing is explained in details, including  SQL for creating View mimicking "information_schema.columns" 
2) Update Phalcon Dialect and Adapter for PostgreSQL so it uses modyfied SQL (modified "information_schema.columns" View SQL) to recognize materialized views.

If anyone needs that, I have time in the next 1-2 months to make contribution to Phalcon, and create 1) or 2).

If nobody needs that, I don't want to waste my time :)

**Comment by BeMySlaveDarlin** [comment]
There is incubator-db repo, feel free to contribute your dialect with docs provided.
